examples/ts-fixture/src/main.ts

```diff
@@ -0,0 +1,6 @@
+import { wing } from "ts4wing";
+import { cloud } from "@winglang/sdk";
+
+wing((app) => {
```

Comment:
What's the benefit of using a function here over the more conventional App class?

I am okay to try this but it feels a bit weird and I am not sure exactly what's the motivation of breaking the CPM specifically for the app.

Reply:
If you look at an existing preflight.js, you'll see:

```js
// ... stuff

class $Root extends $stdlib.std.Resource {
  constructor($scope, $id) {
    super($scope, $id);
    // the actual app goes here, using `this` as a scope
  }
}

const $PlatformManager = new $stdlib.platform.PlatformManager({platformPaths: $platforms});
const $APP = $PlatformManager.createApp({ outdir: $outdir, name: "test", rootConstruct: $Root, isTestEnvironment: $wing_is_test, entrypointDir: process.env['WING_SOURCE_DIR'], rootId: process.env['WING_ROOT_ID'] });
$APP.synth();
```

The important part here is that constructs are by default created in the $Root construct scope rather than the app. This assumption is used elsewhere, like in testing to duplicate the root. I couldn't figure out a way to do this with `new App()`. Even if that `App` is not actually an App but a child of one, we would still need to execute all the code in the entire TS code to duplicate stuff. This felt icky and not worth keeping the constructor-based model for. 

The function is a clear signal that "anything inside here has a lifecycle managed by wing"